### PR TITLE
Modular ingredient handlers

### LINKED ISSUE #337: Organize linter ingredient handlers as modules
A follow up to #320

Complex ingredient handlers don't really work as inline functions in [`ingredient-handlers.js`](https://github.com/mdn/stumptown-content/blob/master/scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers.js). The large ones (like `data.specifications` or `prose.short_description`) ought to be their own modules that are `require`d in `ingredient-handlers.js`.

A structure like this might work:

- `ingredient-handers/`
  - `data-specifications.js`
  - `index.js`
  - `prose-short-description.js`
  - `utils.js`

### PR DESCRIPTION
**Pull request by wbamberg**
This attempts to fix https://github.com/mdn/stumptown-content/issues/337.

Two main commits:
* https://github.com/mdn/stumptown-content/commit/e658eb2c51bb7f11de0ab02a6245260dd3a77687 factors out utility functions, which are expected to be used by >1 handler, into a separate module "utils.js". This also pushes `source` into the `context` object, because we need it in various places.
* https://github.com/mdn/stumptown-content/commit/ec2a79ddad18bc8ba05b61abe57418eb3355c0d7 factors out the three big ingredient handlers into their own modules, and also moves "ingredient-handlers.js" -> "ingredient-handlers/index.js" so everything can live in there.

### CODE REVIEW

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/index.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 const fs = require("fs");
 const path = require("path");
-
 const yaml = require("js-yaml");
```
Comment (ddbeck) [inline scripts/scraper-ng/rules/html-require-recipe-ingredients/index.js]: Side note here: I've been grouping imports by standard library, external libraries, and then local modules. I don't know if we should have a style for this sort of thing, but that's what I was doing.

**Review by ddbeck — APPROVED** [review]
This looks good to me! Thank you, Will.

(I wasn't sure if maybe @Elchi3 also wanted a chance to review this, so I'm going to leave it unmerged for now. Feel free to merge this if it's blocking something else.)

File: scripts/scraper-ng/rules/html-require-recipe-ingredients/index.js (hunk 1 above)
Comment (wbamberg) [reply scripts/scraper-ng/rules/html-require-recipe-ingredients/index.js]: Oh I didn't realise that, but I'm happy for us to to this. I've put the blank line back :).

**Review by Elchi3 — APPROVED** [review]
Thank you, Will! :+1:

**Pull request merged**